content/cookbook_wrapper.md

```diff
@@ -0,0 +1,145 @@
+    cd my_supermarket_wrapper
+    ```
+
+## Define Wrapper Dependencies
+
+3. Defines the wrapper cookbook's dependency on the `SOMETHING` cookbook. Open the metadata.rb file of the new cookbook, and then add the following line:
+
+    ```ruby
+    depends 'SOMETHING'
+    ```
+
+4. Save and close the metadata.rb file.
+
+5. Open the `/recipes/default.rb` recipe located within the new cookbook and add the following content:
+
+    ```ruby
+    include_recipe 'SOMETHING'
+    ```
+
+    This ensures that the `default.rb` file in the `SOMETHING` is run.
+
+## Define Attributes
+
+Define the attributes for the Chef Supermarket installation and how it connects to the Chef Infra Server. One approach is to hard-code attributes in the wrapper cookbook's `default.rb` recipe. A better approach is to place these attributes in a data bag, and then reference them from the recipe. For example, the data bag could be named `apps` and then a data bag item within the data bag could be named `supermarket`.
+
+The following attribute values must be defined:
+
+- `chef_server_url`
+- `chef_oauth2_app_id`
+- `chef_oauth2_secret`
+
+Once configured, you can get the `chef_oauth2_app_id` and `chef_oauth2_secret` values from your Chef Infra Server within `/etc/opscode/oc-id-applications/supermarket.json`:
+
+For `chef_server_url`, enter in the url for your chef server. For `chef_oauth2_app_id`, enter in the uid from `/etc/opscode/oc-id-applications/supermarket.json` For `chef_oauth2_secret`, enter in the secret from `/etc/opscode/oc-id-applications/supermarket.json`
+
+To define these attributes, do the following:
+
+1. Open the `/recipes/default.rb` file and add the following, BEFORE the `include_recipe` line that was added in the previous step. This example uses a data bag named `apps` and a data bag item named `supermarket`:
+
+    ```ruby
+    app = data_bag_item('apps', 'supermarket')
+    ```
+
+2. Set the attributes from the data bag:
+
+    ```ruby
+    node.override['supermarket_omnibus']['chef_server_url'] = app['chef_server_url']
+    node.override['supermarket_omnibus']['chef_oauth2_app_id'] = app['chef_oauth2_app_id']
+    node.override['supermarket_omnibus']['chef_oauth2_secret'] = app['chef_oauth2_secret']
+    ```
+
+    When finished, the `/recipes/default.rb` file should have code similar to:
+
+    ```ruby
+    app = data_bag_item('apps', 'supermarket')
+
+    node.override['supermarket_omnibus']['chef_server_url'] = app['chef_server_url']
+    node.override['supermarket_omnibus']['chef_oauth2_app_id'] = app['chef_oauth2_app_id']
+    node.override['supermarket_omnibus']['chef_oauth2_secret'] = app['chef_oauth2_secret']
+
+    include_recipe 'SOMETHING'
+    ```
+
+3. Save and close the `/recipes/default.rb` file.
+
+{{< note >}}
+
+If you are running your private Supermarket in AWS, you may need to set an additional attribute for the node's public IP address:
+
+```ruby
+node.override['supermarket_omnibus']['config']['fqdn'] = your_node_public_ip
```

Comment:
Since we've already said we're talking about AWS we could use the cloud attribute here.